blockmanager.go

```diff
@@ -1880,61 +1847,30 @@ func (b *blockManager) handleNotifyMsg(notification *blockchain.Notification) {
 		// transaction is still valid.
 		if !headerApprovesParent(&block.MsgBlock().Header) {
 			for _, tx := range parentBlock.Transactions()[1:] {
-				_, err := txMemPool.MaybeAcceptTransaction(tx, false, true)
+				_, err := b.cfg.TxMemPool.MaybeAcceptTransaction(tx, false, true)
 				if err != nil && !isDoubleSpendOrDuplicateError(err) {
-					txMemPool.RemoveTransaction(tx, true)
+					b.cfg.TxMemPool.RemoveTransaction(tx, true)
 				}
 			}
 		}
 
-		if r := b.server.rpcServer; r != nil {
-			// Filter and update the rebroadcast inventory.
-			b.server.PruneRebroadcastInventory()
-
-			// Notify registered websocket clients of incoming block.
-			r.ntfnMgr.NotifyBlockConnected(block)
-		}
-
-		if b.server.bg != nil {
-			b.server.bg.BlockConnected(block)
-		}
-
-	// Stake tickets are spent or missed from the most recently connected block.
-	case blockchain.NTSpentAndMissedTickets:
-		tnd, ok := notification.Data.(*blockchain.TicketNotificationsData)
-		if !ok {
-			bmgrLog.Warnf("Tickets connected notification is not " +
-				"TicketNotificationsData")
-			break
-		}
-
-		if r := b.server.rpcServer; r != nil {
-			r.ntfnMgr.NotifySpentAndMissedTickets(tnd)
-		}
-
-	// Stake tickets are matured from the most recently connected block.
-	case blockchain.NTNewTickets:
-		tnd, ok := notification.Data.(*blockchain.TicketNotificationsData)
-		if !ok {
-			bmgrLog.Warnf("Tickets connected notification is not " +
-				"TicketNotificationsData")
-			break
-		}
+		// Filter and update the rebroadcast inventory.
+		b.cfg.PruneRebroadcastInventory()
```

Comment:
This requires the check if the `rpcServer` is not nil since it only applies when the RPC server is enabled.  There can be nothing to rebroadcast unless it was submitted via the RPC server.